### DISCUSSION TITLE
Create hi-res matrix for image generation

### DISCUSSION THREAD

**Discussion by szamaq** [TheLastBen/fast-stable-diffusion#1896] [Q&A]
Hello Dear Community :)

I am rather new in SD so I hope I can get one thing clarified here. 
When we create image we use ckpt files. Some of them are small and others are 8GB and bigger. The problem is these files are made with 512x512 matrix so if I try to generate native 2048x2048 (upscaled to 8kx8k or not) i always see dozens of hands, legs and heads. I undesrand this is because engine tries to fill all 512x512 blocks in native image with something.
FINE.
What about creating 2048x2048 ckpt files instead? It should work right?
I know this will be more resources consuming but outcome should be totally different. Sometimes (super rarely) I get lucky and generate what I asked for. The quality of picture and details are outstanding. 

When I check Dreambooth I see there is just 512x512 option. Do you know how to create highe resolution matrix so hires images would be generated natively not just from 512x512 but 1kx1k, 1.5k1.5k or most preferable 2kx2k?

I know this will take more vram and time, I can rent a server.

**Comment by Omenizer** [comment]
You can create perfect hi-res images using hires fix or in img2img using SD upscale, no need for a new ckpt

**Comment by szamaq** [comment]
Thank you for answer. This would work but this is still "only" upscaled. I know it looks nice on smartphone where there is 6" screen but on 4k tv it is a mess. Image rendered natively looks way better and who said i cannot use hires fix to that image afterall? This is why I wanted to make high resolution ckpt file.

**Comment by szamaq** [comment]
I just put this here: https://prnt.sc/3YA3zU1srq8r

This is 1024x1024 + hires made from 512x512. And this is just 2x bigger. When I do 4x bigger (2048x2048) this is just waste of energy and server time. That`s why I want to have higher resolution in ckpt files. 

edit:

Guidance Scale: 10, Model: uberrealisticDreamy_pfg, VAE: vae-ft-mse-840000-ema-pruned, Prompt Strength: 0.8, Fix Faces: GFPGANv1.3, Upscale: RealESRGAN_x4plus (4x)

Prompt: sexy girl wearing shiny leggings and sport shoes, horny and hot gaze and look, cute and beautiful very detailed face, (sitting), highly detailed, realistic, anatomical, best quality

Negative Prompt: extra arms, poorly rendered, (((extra legs))), ((extra limbs)), (deformed), ((ugly)), mutilated, disfigured, text, extra fingers, poorly drawn face, mutation, bad proportions, lowres, ((duplicate)), mutilated, ((malformed limbs)), ((deformed body features))


Even if there are suggested in internet negative prompts it doesn`t work with images bigger than native 512x512.

**Comment by Omenizer** [comment]
You're not using this correctly. Hires fix and SD Upscale will both use Stable Diffusion to create the higher res image. Just look at some gens.

![image](https://user-images.githubusercontent.com/119709270/228491727-3ba2771b-f3e2-4756-9bb7-3bc40b8609f0.png)


https://civitai.com/gallery/100657?modelId=8730&modelVersionId=10301&infinite=false&returnUrl=%2Fmodels%2F8730%2Fhipoly-3d-model-lora

**Comment by szamaq** [comment]
According to your photo from from civitai.com:
This is what I am talking about - low res. Please look at details, this is horror. Her hairs, eyes, everything... It looks very low detailed.

Mine is low detailed too, just check her eyes and hairs too. And other things. This was made 1024x1024 natively (upscaled to 4kx4k) from 512x512 ckpt model. This is why I want to make model in high resolution because as you see 512x512 is way too low..

![wearing_shiny__pink__shiny_leggings_CAD7TJ80](https://user-images.githubusercontent.com/56153238/228498604-a9e155df-0918-486e-81e3-dfcad943f615.jpeg)
